PF/Assignment2/Assign 06-08/Day2AssignmentII.java

```diff
@@ -0,0 +1,70 @@
+
+public class Day2AssignmentII {
+	
+	String[] printPyramid(int noOfRows){
+		//checking for valid integer value
+		String[] err=checkForError(noOfRows);
+		if(!err[0].equals("NP"))
+		return err;
+		
+		int N=noOfRows;  
+		String[] pyStrings=new String[2*N-1]; //array to be returned
+		
+		for(int i=0;i<2*N-1;i++)
+		pyStrings[i]=spaces(i+1, N)+numbers(i+1,N); //Decomposition-- getting spaces and number seperately 
+		
+		return pyStrings;
+	}
+	
+	String[] checkForError(int n)
+	{
+		String[] errorMessage=new String[1];
+		errorMessage[0]=new String();
+		errorMessage[0]="NP";
+		
+		if(n<0)
+			{
+			errorMessage[0]="Number cant be negative";
+			return errorMessage;
+			}
+		else if (n==0) {
+			errorMessage[0]="Number cant be Zero";
+			return errorMessage;
+		}
+		else if (n>9) {
+			errorMessage[0]="Number should be a single digit number";
+			return errorMessage;
+		}
+		return errorMessage;
+	}
+	String spaces(int row,int n){
```

Comment:
make the method public so that it is visible in every package. Use it default if there is a requirement.